### ISSUE TITLE
ImportError: No module named exceptions

### ISSUE THREAD

**Issue by vertangelx** [ipython/ipython#9065]
After installing Anaconda (64bit) on OSX 10.10 and trying to start ipython notebook via `ipython notebook`, I get the error below. The same problem exists just running `ipython`

```
$ ipython notebook
Traceback (most recent call last):
  File "/Users/vertangelx/anaconda/envs/test/bin/ipython", line 4, in <module>
    from IPython import start_ipython
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/__init__.py", line 47, in <module>
    from .core.application import Application
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/core/application.py", line 24, in <module>
    from IPython.core import release, crashhandler
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/core/crashhandler.py", line 28, in <module>
    from IPython.core import ultratb
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/core/ultratb.py", line 116, in <module>
    from IPython.utils import path as util_path
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/utils/path.py", line 19, in <module>
    from IPython.utils.process import system
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/utils/process.py", line 19, in <module>
    from ._process_posix import system, getoutput, arg_split, check_pid
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/IPython/utils/_process_posix.py", line 24, in <module>
    import pexpect
  File "/Users/vertangelx/anaconda/envs/test/lib/python2.7/site-packages/pexpect/__init__.py", line 69, in <module>
    from .exceptions import ExceptionPexpect, EOF, TIMEOUT
ImportError: No module named exceptions
```

Why is this error happening on this particular system?

Below is the list of packages in my virtualenv

```
abstract-rendering        0.5.1               np110py27_0    defaults
alabaster                 0.7.6                    py27_0    defaults
anaconda                  2.4.0               np110py27_0    defaults
anaconda-client           1.1.0                    py27_0    defaults
appnope                   0.1.0                    py27_0    defaults
appscript                 1.0.1                    py27_0    <unknown>
argcomplete               1.0.0                    py27_1    defaults
astropy                   1.0.5               np110py27_1    defaults
babel                     2.1.1                    py27_0    defaults
backports.ssl-match-hostname 3.4.0.2                   <pip>
beautifulsoup4            4.4.1                    py27_0    defaults
bitarray                  0.8.1                    py27_0    <unknown>
blaze                     0.8.3                     <pip>
blaze-core                0.8.3                    py27_0    defaults
bokeh                     0.10.0                   py27_0    defaults
boto                      2.38.0                   py27_0    <unknown>
bottleneck                1.0.0               np110py27_0    defaults
cdecimal                  2.3                      py27_0    <unknown>
cffi                      1.2.1                    py27_0    defaults
clyent                    0.4.0                    py27_0    defaults
colorama                  0.3.3                    py27_0    <unknown>
configobj                 5.0.6                    py27_0    <unknown>
cryptography              1.0.2                    py27_0    defaults
curl                      7.45.0                        0    defaults
cython                    0.23.4                   py27_0    defaults
cytoolz                   0.7.4                    py27_0    defaults
datashape                 0.4.7               np110py27_1    defaults
decorator                 4.0.4                    py27_0    defaults
docutils                  0.12                     py27_0    <unknown>
enum34                    1.0.4                    py27_0    <unknown>
fastcache                 1.0.2                    py27_0    <unknown>
flask                     0.10.1                   py27_1    <unknown>
freetype                  2.5.5                         0    defaults
funcsigs                  0.4                      py27_0    <unknown>
gevent                    1.0.1                    py27_0    <unknown>
gevent-websocket          0.9.3                    py27_0    <unknown>
greenlet                  0.4.9                    py27_0    defaults
grin                      1.2.1                    py27_1    <unknown>
h5py                      2.5.0               np110py27_4    defaults
hdf5                      1.8.15.1                      2    defaults
idna                      2.0                      py27_0    <unknown>
ipaddress                 1.0.14                   py27_0    defaults
ipykernel                 4.1.1                    py27_0    defaults
ipython                   4.0.0                    py27_1    defaults
ipython-genutils          0.1.0                     <pip>
ipython-notebook          4.0.4                    py27_0    defaults
ipython-qtconsole         4.0.1                    py27_0    defaults
ipython_genutils          0.1.0                    py27_0    defaults
ipywidgets                4.1.0                    py27_0    defaults
itsdangerous              0.24                     py27_0    <unknown>
jdcal                     1.0                      py27_0    <unknown>
jedi                      0.9.0                    py27_0    defaults
jinja2                    2.8                      py27_0    defaults
jpeg                      8d                            1    <unknown>
jsonschema                2.4.0                    py27_0    <unknown>
jupyter                   1.0.0                    py27_0    defaults
jupyter-client            4.1.1                     <pip>
jupyter-console           4.0.3                     <pip>
jupyter-core              4.0.6                     <pip>
jupyter_client            4.1.1                    py27_0    defaults
jupyter_console           4.0.3                    py27_0    defaults
jupyter_core              4.0.6                    py27_0    defaults
launcher                  1.0.0                         3    <unknown>
libpng                    1.6.17                        0    <unknown>
libtiff                   4.0.6                         0    defaults
libxml2                   2.9.2                         0    <unknown>
libxslt                   1.1.28                        2    <unknown>
llvmlite                  0.7.0                    py27_3    defaults
lxml                      3.4.4                    py27_0    <unknown>
markupsafe                0.23                     py27_0    <unknown>
matplotlib                1.4.3               np110py27_3    defaults
mistune                   0.7.1                    py27_0    defaults
multipledispatch          0.4.8                    py27_0    defaults
nbconvert                 4.0.0                    py27_0    defaults
nbformat                  4.0.1                    py27_0    defaults
networkx                  1.10                     py27_0    defaults
nltk                      3.1                      py27_0    defaults
node-webkit               0.10.1                        0    <unknown>
nose                      1.3.7                    py27_0    <unknown>
notebook                  4.0.6                    py27_0    defaults
numba                     0.21.0              np110py27_0    defaults
numexpr                   2.4.4               np110py27_0    defaults
numpy                     1.10.1                   py27_0    defaults
odo                       0.3.4                    py27_0    defaults
openpyxl                  2.2.6                    py27_0    defaults
openssl                   1.0.2d                        0    defaults
pandas                    0.17.0              np110py27_0    defaults
path.py                   8.1.2                    py27_0    defaults
patsy                     0.4.0               np110py27_0    defaults
pep8                      1.6.2                    py27_0    <unknown>
pexpect                   3.3                      py27_0    defaults
pickleshare               0.5                      py27_0    defaults
pillow                    3.0.0                    py27_1    defaults
pip                       7.1.2                    py27_0    defaults
ply                       3.8                      py27_0    defaults
psutil                    3.2.2                    py27_0    defaults
ptyprocess                0.5                      py27_0    defaults
py                        1.4.30                   py27_0    defaults
pyasn1                    0.1.9                    py27_0    defaults
pyaudio                   0.2.7                    py27_0    <unknown>
pycosat                   0.6.1                    py27_0    <unknown>
pycparser                 2.14                     py27_0    <unknown>
pycrypto                  2.6.1                    py27_0    <unknown>
pycurl                    7.19.5.1                 py27_3    defaults
pyflakes                  1.0.0                    py27_0    defaults
pygments                  2.0.2                    py27_0    <unknown>
pyopenssl                 0.15.1                   py27_1    <unknown>
pyparsing                 2.0.3                    py27_0    <unknown>
pyqt                      4.11.4                   py27_0    defaults
pytables                  3.2.2               np110py27_0    defaults
pytest                    2.8.1                    py27_0    defaults
python                    2.7.10                        2    defaults
python-dateutil           2.4.2                    py27_0    <unknown>
python.app                1.2                      py27_4    <unknown>
pytz                      2015.6                   py27_0    defaults
pyyaml                    3.11                     py27_1    <unknown>
pyzmq                     14.7.0                   py27_1    defaults
qt                        4.8.7                         1    defaults
qtconsole                 4.1.0                    py27_0    defaults
readline                  6.2                           2    <unknown>
redis                     2.6.9                         0    <unknown>
redis-py                  2.10.3                   py27_0    <unknown>
requests                  2.8.1                    py27_0    defaults
rope                      0.9.4                    py27_1    <unknown>
scikit-image              0.11.3              np110py27_0    defaults
scikit-learn              0.16.1              np110py27_0    defaults
scipy                     0.16.0              np110py27_1    defaults
setuptools                18.4                     py27_0    defaults
simplegeneric             0.8.1                    py27_0    defaults
singledispatch            3.4.0.3                  py27_0    defaults
sip                       4.16.9                   py27_0    defaults
six                       1.10.0                   py27_0    defaults
snowballstemmer           1.2.0                    py27_0    <unknown>
sockjs-tornado            1.0.1                    py27_0    <unknown>
sphinx                    1.3.1                    py27_0    <unknown>
sphinx-rtd-theme          0.1.7                     <pip>
sphinx_rtd_theme          0.1.7                    py27_0    <unknown>
spyder                    2.3.7                    py27_1    defaults
spyder-app                2.3.7                    py27_0    defaults
sqlalchemy                1.0.9                    py27_0    defaults
sqlite                    3.8.4.1                       1    <unknown>
ssl_match_hostname        3.4.0.2                  py27_0    <unknown>
statsmodels               0.6.1               np110py27_0    defaults
sympy                     0.7.6.1                  py27_0    defaults
tables                    3.2.2                     <pip>
terminado                 0.5                      py27_1    defaults
tk                        8.5.18                        0    <unknown>
toolz                     0.7.4                    py27_0    defaults
tornado                   4.2.1                    py27_1    defaults
traitlets                 4.0.0                    py27_0    defaults
ujson                     1.33                     py27_0    <unknown>
unicodecsv                0.14.1                   py27_0    defaults
werkzeug                  0.10.4                   py27_0    <unknown>
wheel                     0.26.0                   py27_1    defaults
xlrd                      0.9.4                    py27_0    defaults
xlsxwriter                0.7.7                    py27_0    defaults
xlwings                   0.4.1                    py27_0    defaults
xlwt                      1.0.0                    py27_0    <unknown>
yaml                      0.1.6                         0    <unknown>
zeromq                    4.1.3                         0    defaults
zlib                      1.2.8                         0    <unknown>
```

**Comment by takluyver** [comment]
It looks like you've got mixed code from two different versions of pexpect - it claims Pexpect 3.3 is installed, but that import is from Pexpect 4.

I don't know how that could have happened, but try:

```
conda remove pexpect
conda clean --tarballs --packages
conda install pexpect
```

You may need to install Jupyter again after that, if conda uninstalled it along with pexpect.

**Comment by vertangelx** [comment]
@takluyver Tried your suggestion, the same error persists running either `ipython` or `ipython notebook`. I did not try to install Jupyter again.

Here's my `conda list` after the suggested remove/clean/install. Looks like it's still Pexpect 3.3. Should we force it to install a specific version of Pexpect?

	abstract-rendering        0.5.1               np110py27_0    defaults
	alabaster                 0.7.6                    py27_0    defaults
	anaconda                  2.4.0               np110py27_0    defaults
	anaconda-client           1.1.0                    py27_0    defaults
	appnope                   0.1.0                    py27_0    defaults
	appscript                 1.0.1                    py27_0    <unknown>
	argcomplete               1.0.0                    py27_1    defaults
	astropy                   1.0.5               np110py27_1    defaults
	babel                     2.1.1                    py27_0    defaults
	backports.ssl-match-hostname 3.4.0.2                   <pip>
	beautifulsoup4            4.4.1                    py27_0    defaults
	bitarray                  0.8.1                    py27_0    <unknown>
	blaze                     0.8.3                     <pip>
	blaze-core                0.8.3                    py27_0    defaults
	bokeh                     0.10.0                   py27_0    defaults
	boto                      2.38.0                   py27_0    <unknown>
	bottleneck                1.0.0               np110py27_0    defaults
	cdecimal                  2.3                      py27_0    <unknown>
	cffi                      1.2.1                    py27_0    defaults
	clyent                    0.4.0                    py27_0    defaults
	colorama                  0.3.3                    py27_0    <unknown>
	configobj                 5.0.6                    py27_0    <unknown>
	cryptography              1.0.2                    py27_0    defaults
	curl                      7.45.0                        0    defaults
	cython                    0.23.4                   py27_0    defaults
	cytoolz                   0.7.4                    py27_0    defaults
	datashape                 0.4.7               np110py27_1    defaults
	decorator                 4.0.4                    py27_0    defaults
	docutils                  0.12                     py27_0    <unknown>
	enum34                    1.0.4                    py27_0    <unknown>
	fastcache                 1.0.2                    py27_0    <unknown>
	flask                     0.10.1                   py27_1    <unknown>
	freetype                  2.5.5                         0    defaults
	funcsigs                  0.4                      py27_0    <unknown>
	gevent                    1.0.1                    py27_0    <unknown>
	gevent-websocket          0.9.3                    py27_0    <unknown>
	greenlet                  0.4.9                    py27_0    defaults
	grin                      1.2.1                    py27_1    <unknown>
	h5py                      2.5.0               np110py27_4    defaults
	hdf5                      1.8.15.1                      2    defaults
	idna                      2.0                      py27_0    <unknown>
	ipaddress                 1.0.14                   py27_0    defaults
	ipykernel                 4.1.1                    py27_0    defaults
	ipython                   4.0.0                    py27_1    defaults
	ipython-genutils          0.1.0                     <pip>
	ipython-notebook          4.0.4                    py27_0    defaults
	ipython-qtconsole         4.0.1                    py27_0    defaults
	ipython_genutils          0.1.0                    py27_0    defaults
	ipywidgets                4.1.0                    py27_0    defaults
	itsdangerous              0.24                     py27_0    <unknown>
	jdcal                     1.0                      py27_0    <unknown>
	jedi                      0.9.0                    py27_0    defaults
	jinja2                    2.8                      py27_0    defaults
	jpeg                      8d                            1    <unknown>
	jsonschema                2.4.0                    py27_0    <unknown>
	jupyter                   1.0.0                    py27_0    defaults
	jupyter-client            4.1.1                     <pip>
	jupyter-console           4.0.3                     <pip>
	jupyter-core              4.0.6                     <pip>
	jupyter_client            4.1.1                    py27_0    defaults
	jupyter_console           4.0.3                    py27_0    defaults
	jupyter_core              4.0.6                    py27_0    defaults
	launcher                  1.0.0                         3    <unknown>
	libpng                    1.6.17                        0    <unknown>
	libtiff                   4.0.6                         0    defaults
	libxml2                   2.9.2                         0    <unknown>
	libxslt                   1.1.28                        2    <unknown>
	llvmlite                  0.7.0                    py27_3    defaults
	lxml                      3.4.4                    py27_0    <unknown>
	markupsafe                0.23                     py27_0    <unknown>
	matplotlib                1.4.3               np110py27_3    defaults
	mistune                   0.7.1                    py27_0    defaults
	multipledispatch          0.4.8                    py27_0    defaults
	nbconvert                 4.0.0                    py27_0    defaults
	nbformat                  4.0.1                    py27_0    defaults
	networkx                  1.10                     py27_0    defaults
	nltk                      3.1                      py27_0    defaults
	node-webkit               0.10.1                        0    <unknown>
	nose                      1.3.7                    py27_0    <unknown>
	notebook                  4.0.6                    py27_0    defaults
	numba                     0.21.0              np110py27_0    defaults
	numexpr                   2.4.4               np110py27_0    defaults
	numpy                     1.10.1                   py27_0    defaults
	odo                       0.3.4                    py27_0    defaults
	openpyxl                  2.2.6                    py27_0    defaults
	openssl                   1.0.2d                        0    defaults
	pandas                    0.17.0              np110py27_0    defaults
	path.py                   8.1.2                    py27_0    defaults
	patsy                     0.4.0               np110py27_0    defaults
	pep8                      1.6.2                    py27_0    <unknown>
	pexpect                   3.3                      py27_0    defaults
	pickleshare               0.5                      py27_0    defaults
	pillow                    3.0.0                    py27_1    defaults
	pip                       7.1.2                    py27_0    defaults
	ply                       3.8                      py27_0    defaults
	psutil                    3.2.2                    py27_0    defaults
	ptyprocess                0.5                      py27_0    defaults
	py                        1.4.30                   py27_0    defaults
	pyasn1                    0.1.9                    py27_0    defaults
	pyaudio                   0.2.7                    py27_0    <unknown>
	pycosat                   0.6.1                    py27_0    <unknown>
	pycparser                 2.14                     py27_0    <unknown>
	pycrypto                  2.6.1                    py27_0    <unknown>
	pycurl                    7.19.5.1                 py27_3    defaults
	pyflakes                  1.0.0                    py27_0    defaults
	pygments                  2.0.2                    py27_0    <unknown>
	pyopenssl                 0.15.1                   py27_1    <unknown>
	pyparsing                 2.0.3                    py27_0    <unknown>
	pyqt                      4.11.4                   py27_0    defaults
	pytables                  3.2.2               np110py27_0    defaults
	pytest                    2.8.1                    py27_0    defaults
	python                    2.7.10                        2    defaults
	python-dateutil           2.4.2                    py27_0    <unknown>
	python.app                1.2                      py27_4    <unknown>
	pytz                      2015.6                   py27_0    defaults
	pyyaml                    3.11                     py27_1    <unknown>
	pyzmq                     14.7.0                   py27_1    defaults
	qt                        4.8.7                         1    defaults
	qtconsole                 4.1.0                    py27_0    defaults
	readline                  6.2                           2    <unknown>
	redis                     2.6.9                         0    <unknown>
	redis-py                  2.10.3                   py27_0    <unknown>
	requests                  2.8.1                    py27_0    defaults
	rope                      0.9.4                    py27_1    <unknown>
	scikit-image              0.11.3              np110py27_0    defaults
	scikit-learn              0.16.1              np110py27_0    defaults
	scipy                     0.16.0              np110py27_1    defaults
	setuptools                18.5                     py27_0    defaults
	simplegeneric             0.8.1                    py27_0    defaults
	singledispatch            3.4.0.3                  py27_0    defaults
	sip                       4.16.9                   py27_0    defaults
	six                       1.10.0                   py27_0    defaults
	snowballstemmer           1.2.0                    py27_0    <unknown>
	sockjs-tornado            1.0.1                    py27_0    <unknown>
	sphinx                    1.3.1                    py27_0    <unknown>
	sphinx-rtd-theme          0.1.7                     <pip>
	sphinx_rtd_theme          0.1.7                    py27_0    <unknown>
	spyder                    2.3.7                    py27_1    defaults
	spyder-app                2.3.7                    py27_0    defaults
	sqlalchemy                1.0.9                    py27_0    defaults
	sqlite                    3.8.4.1                       1    <unknown>
	ssl_match_hostname        3.4.0.2                  py27_0    <unknown>
	statsmodels               0.6.1               np110py27_0    defaults
	sympy                     0.7.6.1                  py27_0    defaults
	tables                    3.2.2                     <pip>
	terminado                 0.5                      py27_1    defaults
	tk                        8.5.18                        0    <unknown>
	toolz                     0.7.4                    py27_0    defaults
	tornado                   4.2.1                    py27_1    defaults
	traitlets                 4.0.0                    py27_0    defaults
	ujson                     1.33                     py27_0    <unknown>
	unicodecsv                0.14.1                   py27_0    defaults
	werkzeug                  0.10.4                   py27_0    <unknown>
	wheel                     0.26.0                   py27_1    defaults
	xlrd                      0.9.4                    py27_0    defaults
	xlsxwriter                0.7.7                    py27_0    defaults
	xlwings                   0.4.1                    py27_0    defaults
	xlwt                      1.0.0                    py27_0    <unknown>
	yaml                      0.1.6                         0    <unknown>
	zeromq                    4.1.3                         0    defaults
	zlib                      1.2.8                         0    <unknown>

**Comment by vertangelx** [comment]
Additionally, `anaconda search -t conda pexpect` does not show a version newer than 3.3 for osx-64.

	Packages:
	     Name                      |  Version | Package Types   | Platforms      
	     ------------------------- |   ------ | --------------- | ---------------
	     asmeurer/pexpect          |      2.4 | conda           | osx-64         
	                                          : http://pexpect.sourceforge.net/
	     auto/pexpect              |      3.1 | conda           | linux-64, linux-32, osx-64
	                                          : Pexpect allows easy control of interactive console applications.
	     auto/pexpect-nm           |      2.6 | conda           | linux-64       
	                                          : http://pexpect.sourceforge.net/
	     auto/pexpect-u            |    2.5.1 | conda           | linux-64       
	                                          : https://bitbucket.org/takluyver/pexpect
	     mhworth/pexpect           |      3.0 | conda           | linux-64, osx-64
	                                          : Pexpect allows easy control of interactive console applications.
	     pdrops/pexpect            |      3.0 | conda           | linux-64, osx-64
	                                          : Pexpect allows easy control of interactive console applications.
	     severson/pexpect          |      2.4 | conda           | osx-64         
	                                          : http://pexpect.sourceforge.net/
	     ska/pexpect               |      3.3 | conda           | linux-64, osx-64
	                                          : Pexpect allows easy control of interactive console applications.
	     travis/pexpect            |      2.4 | conda           | osx-64         
	                                          : http://pexpect.sourceforge.net/
	     xanderhsia/pexpect        |      3.3 | conda           | osx-64         
	                                          : Pexpect allows easy control of interactive console applications.

**Comment by takluyver** [comment]
If it was really Pexpect 3.3, that would be OK. But that import line is [not in 3.3](https://github.com/pexpect/pexpect/blob/3.3/pexpect/__init__.py). A copy of Pexpect 4 must have got mixed up in there somewhere, but I don't know where.

**Comment by rdhyee** [comment]
I just ran into this problem while installing `jupyter` in a new conda env on Mac OS X 10.9.5.  

After doing:
```
conda create -n ipyw-test python=2 pip
. activate ipyw-test
conda install jupyter
```
I got an error running `ipython` because of the missing `exceptions` package.  I used the [suggested fix](https://github.com/ipython/ipython/issues/9065#issuecomment-162163512):

```
conda remove pexpect
conda clean --tarballs --packages
conda install pexpect
```
but then had to add

```
pip install -U pexpect
```

because conda didn't have the correct version of `pexpect`.

I'll report the issue to <https://github.com/conda/conda/issues>.
